Yank/utils.py

```diff
@@ -252,6 +252,66 @@ def _wrapper(*args, **kwargs):
 
 
 # =======================================================================================
+# Multimethod wrapping to better handle cyclomatic complexity with input types
+# See Adam Bard's post: https://adambard.com/blog/implementing-multimethods-in-python/
```

Comment:
This is pretty cool! But I don't understand the difference with the `functools.singledispatch()` that is in the standard library.

Reply:
While that is a viable solution in normal functions, the problem is that `singledispatch` does not work on class methods, since the first argument is always `self` and not the arguments being passed as the first arg to run type checking on. The multi method is a bit more general, albeit requires to be implemented as its own methods

Reply:
Ah, I understand now, thanks! Very cool! What do you think about using a wrapper like this instead: https://stackoverflow.com/a/24602374 This way we could have a single wrapper and keep the usual `singledispatch` syntax.

Reply:
Hmmm. I saw that and wondered how general it is. I'll give it a try and see what comes of it.